.gitlab-ci.yml

```diff
@@ -22,22 +19,18 @@ variables:
 .build:                            &docker_build
   stage:                           build
   image:                           docker:stable
-  only:
-    refs:
-      - schedules
-      - web
-    kubernetes:                    active
+  rules:
+    # same as only: variables: $x
+    - if:                          '$DOCKERIMAGE == $CI_JOB_NAME'
+    # triggers rebuild when Dockerfile is changed
+    - changes:
+      - 'dockerfiles/${IMAGE_NAME}/Dockerfile'
   tags:
     - kubernetes-parity-build
   services:
     - docker:dind
   environment:
     name:                          parity-build
-  before_script:
-    - test -z "$DOCKERIMAGE" && echo "DOCKERIMAGE must be defined" && exit 1
```

Comment:
@gabreal FYI, I'm removing this since CI is triggered only by $DOCKERFILES variable anyway